### ISSUE TITLE
SplitButton: accessibility

### ISSUE THREAD

**Issue by adrianfiedler** [primefaces/primefaces#3020]
## 1) Environment
- PrimeFaces version: Primefaces 6.1
- Does it work on the newest released PrimeFaces version? No
- Does it work on the newest sources in GitHub? No
- Application server + version: JBoss EAP 7.0
- Affected browsers: Firefox ESR 52.5.0 (64-Bit)
- Screenreader: JAWS 18.0

## 2) Expected behavior
<p:splitButton> main button should be read like a standard button. Dropdown button should be read as "Dropdown". Dropdown menu should be accessible only by keyboard.

## 3) Actual behavior
<p:splitButton> main button is read as something like "UI button switch". Dropdown toggler is not read as "Dropdown". Also I can not select any of the dropdown values with keyboard because the dropdown is immediately closed.

## 4) Steps to reproduce
Read <p:splitButton> component with screenreader

**Comment by melloware** [comment]
If you want to anaylze the code and decide what "role" each HTML component should have.

https://github.com/primefaces/primefaces/blob/master/src/main/java/org/primefaces/component/splitbutton/SplitButtonRenderer.java

**Comment by melloware** [comment]
@MarkusChur Can you help me fix this one?

**Comment by MarkusChur** [comment]
# Fix for Screen readers
the menu button is missing the following attributes:

- `aria-haspopup="true"`: then it is announced as submenu button
- `aria-controls="idOfSubMenu"`: this has to be set to the id of the submenu `<div>` which is toggled. The id is already implemented it just has to be set. 
- `aria-expanded="true|false"`: the state of the menu has to be applied to the button

# Keyboard accessibility
- The submenu is inaccessible for keyboard users. I can't tab into it. It automatically hides when tabbing through. 
- Submenu items seem not be in taborder because the elements are at the very end of the page. The submenu overlay has to be near the the button then it should work.

**Comment by melloware** [comment]
@MarkusChur can you review my PR around the ARIA stuff.  I believe I understood what you are saying.

As for Keyboard Accessbility I can TAB to the button, press SPACEBAR to bring up the menu and then use Arrow up and Down keys to navigate the menu.  Pressing TAB again moves you to the next control?  I don't think any of the PrimeFaces menus allow TAB for navigation its always with the arrow keys.

**Comment by MarkusChur** [comment]
looks good!

@melloware regarding keyboard control. Yes, it can be either arrow keys or tabs. But if arrow keys are used the focus has to jump to the first element in the list and the `tabindex="-1"` has to be set for the first element in the list. 

Please see https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html
There you can transfer all necessary attributes. The table "Menu Button" is covered. Please take a closer look at "Menu". Thank you!

**Comment by melloware** [comment]
OK I read through that spec and added TabIndex=-1 on the Menu and each "A" hyperlink which is what their example is doing.

Ours is slightly different than theirs as the role="menu" is on our container DIV and not the UL. Can you review again I think the -1's will do it.

**Comment by MarkusChur** [comment]
I tested this with the master snapshot and NVDA, there are still issues.

- no focus is set on first menu element when toggling the button
- menu is not announced properly. I think the `<ul> `needs `role="menu"` and `<li>` needs `role="none"`, and `<a> `needs `role="menuitem"`. Even if I fix it with chrome developer console it does not count correctly. This might be related to the wrapping elements.

**Comment by melloware** [comment]
Thanks for testing.  Do you have suggestions that will make it work?

**Comment by MarkusChur** [comment]
Yes, but via PR. Please bear with me that this takes a bit :) Also in regards to the other PRs I wanted to provide.

**Comment by melloware** [comment]
No worries.  I appreciate the help you have provided!  I reopened this issue so we can track it.

**Comment by melloware** [comment]
OK I was able to make the necessary changes to make it match other menus.

UL - gets ARIA_ROLE_MENU
LI - get ARIA_ROLE_NONE
A - get ARIAL_ROLE_MENUITEM
